### DISCUSSION TITLE
Transferring maintenance & keeping pest alive

### DISCUSSION THREAD

**Discussion by CAD97** [pest-parser/pest#606] [General]
- # #664

Unfortunately, I just don't have the bandwidth to drive maintenance/development of pest. I do have just enough admin rights that I can properly give maintenance capabilities to willing community members.

Yes, I said something along these lines [in 2021](https://github.com/pest-parser/pest/discussions/547)... but I think I mean it this time?

## What I can do

| Organization | [Role](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization#permissions-for-organization-roles) |
| - | - |
| [@pest-parser](https://github.com/pest-parser) | Owner |

| Team | Role |
| - | - |
| [@pest-parser/admins](https://github.com/orgs/pest-parser/teams/admins) | Member |
| [@pest-parser/book](https://github.com/orgs/pest-parser/teams/book) | - |
| [@pest-parser/deconstruct](https://github.com/orgs/pest-parser/teams/deconstruct) | Member |
| [@pest-parser/intellij-pest](https://github.com/orgs/pest-parser/teams/intellij-pest) | - |
| [@pest-parser/pest](https://github.com/orgs/pest-parser/teams/pest) | - |
| [@pest-parser/pest-vim](https://github.com/orgs/pest-parser/teams/pest-vim) | - |

| Repository | [Role](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization) |
| - | - |
| [pest-parser/ast](https://github.com/pest-parser/ast) | Write |
| [pest-parser/pest](https://github.com/pest-parser/pest) | Admin |
| [pest-parser/book](https://github.com/pest-parser/book) | - |
| [pest-parser/pest-fmt](https://github.com/pest-parser/pest-fmt) | - |
| [pest-parser/intellij-pest](https://github.com/pest-parser/intellij-pest) | - |
| [pest-parser/pest.vim](https://github.com/pest-parser/pest.vim) | - |
| [pest-parser/site](https://github.com/pest-parser/site) | - |
| [pest-parser/pest3](https://github.com/pest-parser/pest3) | - |
| [pest-parser/site](https://github.com/pest-parser/pest-parser.github.io) | - |

## Publishing

The @pest-parser/admins team has [crates-io publish permission](https://crates.io/teams/github:pest-parser:admins). Organization owners can add trusted users to the group to give them publish permissions.

## Handling the transition

If you are a [contributor](https://github.com/pest-parser/pest/graphs/contributors) already: ask below and I will give you **Maintain** access to pest-parser/pest.

If you ask below and get a second from a contributor, i will give you **Maintain** access.
If you ask below and get a second from anyone, I will give you **Triage** access.

At the end of July, I'll pick the two people added via the above process that seem the most trustworthy and add them as **Owner**s of the organization. From there, it's your project! ❤️

... And if I fail to follow up, please @ me aggressively.

**Comment by jhpratt** [comment]
I have been interested in potentially using `pest` to back the parser for `time`. As such I _might_ be open to this if no one else is. It would require some significant changes before I could use it, though, so I'm not positive.

**Reply by tadman** [reply]
I'd like to help out in some capacity. I've already chipped in some pull requests and would love to see Pest succeed going forward.

**Reply by jhpratt** [reply]
I've just accepted the invitation from @CAD97 to join the organization (and the triage team). I'm still not certain if this is something I will do, but it's better to be a member as the invitation expired after a week. I'm also focused in Rust (stdlib & compiler) and `time` right now. But we'll see :slightly_smiling_face:

**Comment by maksugr** [comment]
I'd like to create a couple of open-source parsers in the near future. And it seems that PEG is the best way for that. So I'm really interested in good being of Pest:) 

I'm a relative newcomer in Rust, but I have quite a lot of experience in other languages and technologies (currently I'm a technical lead of the development team). 

So if I can be useful to Pest, the open-source and rust community I would be proud to help 🙌

**Comment by tomtau** [comment]
Hi everyone!👋
I used `pest` in a few small projects, so I'm naturally interested in keeping it alive and I'd love to help out.❤️
My former academic advisor and colleagues did some [modular parsing](https://i.cs.hku.hk/~bruno/papers/sle17.pdf)-related work, so I'm somewhat familiar with PEGs, but I haven't looked into `pest`'s internal details too much yet. I started to dip my toes into the codebase a bit by [this "good-first-issue" fix](https://github.com/pest-parser/pest/pull/609).

**Comment by nathanielknight** [comment]
I love Pest, and would be heartbroken to see it fade away. While I don't have the background to work directly on the parser I'd be happy to chip in on documentation, testing, maintaining the website, etc.

**Comment by ImJeremyHe** [comment]
I used `pest` in my project. Sorry to hear that and thank for your maintenance.
I think it would be the best that someone fork this repo and keep maintaining.

**Reply by CAD97** [reply]
We're in the progress of getting people maintainer access to this repo and package 🙂 so once that is complete proper maintenance should resume seamlessly.

**Comment by glyn** [comment]
I've been writing an IETF reference implementation in pest. Having grammar files without source code interspersed (unlike some other PEG parsers) makes them easier to read by those who don't necessarily know Rust. So I'd really like pest to continue or a fork to succeed it.

I don't anticipate being able to devote a lot of time on pest, but I'd like to help out here and there to keep the project going.

**Comment by bobbbay** [comment]
I would like to work on pest. I still remember the first time I used pest, 2+ years ago... I would love to chip into maintenance. Specifically, error reporting has been one of my soft spots, so I'd like to continue development on that front (integration with `miette`, for example).

**Comment by yarcowang** [comment]
Em... yeah, I like pest. But I'm new to Rust/pest, I don't think I have enough knowledge to do that and also the time as they said.
But if you need documentation or some lightweight tasks, just @ me.

**Comment by NoahTheDuke** [comment]
I love maintaining stuff, hook me up.
